src/main/kotlin/org/jetbrains/research/ml/kotlinAnalysis/Pipeline.kt

```diff
@@ -0,0 +1,37 @@
+package org.jetbrains.research.ml.kotlinAnalysis
+
+import com.intellij.ide.impl.ProjectUtil
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+
+class Pipeline(private val outputDir: Path) {
+    private val indexer = IndexBuilder()
+    private val psiProvider = PsiProvider()
+    private val adapter = CloneDetectionAdapter()
+
+    fun extractMethodsToCloneDetectionFormat(inputDir: Path) {
+        val projectIndexWriter = File(outputDir.toFile(), "project_index.txt").printWriter()
+        val methodIndexWriter = File(outputDir.toFile(), "method_index.csv").printWriter()
+        val methodDataWriter = File(outputDir.toFile(), "method_data.txt").printWriter()
+
+        Files.walk(inputDir, 1)
```

Comment:
Probably you can create a util function to get all directories from the directory?